cmd/fyne/internal/commands/bundle.go

```diff
@@ -138,35 +138,23 @@ func (b *Bundler) Run(args []string) {
 	}
 }
 
-func (b *Bundler) bundleAction(ctx *cli.Context) error {
+func (b *Bundler) bundleAction(ctx *cli.Context) (err error) {
 	if ctx.Args().Len() != 1 {
 		return errors.New("missing required file or directory parameter after flags")
 	}
 
 	outFile := os.Stdout
 	if b.out != "" {
-		fileModes := os.O_RDWR | os.O_CREATE | os.O_TRUNC
-		if b.noheader {
-			fileModes = os.O_RDWR | os.O_APPEND
+		file, closeFile, err := getOutputFile(b.out, b.noheader)
+		if err != nil {
+			return err
 		}
-
-		f, err := os.OpenFile(b.out, fileModes, 0666)
-		if err == nil {
-			outFile = f
-		} else {
-			if os.IsNotExist(err) {
-				f, err = os.Create(b.out)
-				if err == nil {
-					outFile = f
-				} else {
-					fyne.LogError("Unable to read, or create, output file : "+b.out, err)
-					return err
-				}
-			} else {
-				fyne.LogError("Unable to open output file", err)
-				return err
+		defer func() {
+			if r := closeFile(); r != nil {
+				err = r
```

Comment:
This is not setting the return `err` but the one a few lines above that shadows it.

Reply:
No it doesn't. That is not how named returns and `defer` works.

Reply:
It seems from a little more digging that this is not special for `defer` but for named returns.
Newly assigned variables don't shadow return names like I thought they did.
Apologies for the confusion.